### ISSUE TITLE
Improve the performance of deleting large meshes. 

### ISSUE THREAD

**Issue by theoryshaw** [IfcOpenShell/IfcOpenShell#3226]
Deleting large meshes is painful sometimes.  It's takes a long time, if at all. 

Could this be streamlined somehow?

**Comment by Moult** [comment]
So I'm a little disappointed I had to do this (ping @aothms) but I took a leaf out of @brunopostle 's book and decided to implement a string replacement version of "remove_deep2". The performance difference is just too huge.

In #3244 there are a few chandeliers in there. Some have less polygons and some have more (even though they look identical?). The one with the most polygons has something like 350,000 (!!!) indexed polygonal faces. It takes about half an hour to delete on my machine (which has pretty beefy CPU). This is after the fix in #3052 - which as it turns out has less of an impact when the model is large (e.g. removing 1000 elements on a 10MB file is super fast, but removing 1000 elements on a 200MB file takes about 5 seconds). With this new method, the time taken to delete the outlier chandelier **dropped from half an hour down to only 30 seconds**. (note that @theoryshaw has a few versions of his model with different chandeliers and different model sizes, so careful which model you check with - if you test with a small model, the benefit won't be visible).

Yes, in an ideal scenario, we shouldn't be storing these types of geometry in IFC in the first place, and that model is over 200MB because it has over 500 IfcLightFixtures where a single light bulb has 15,000 polygons and it has little bits of filament modeled ...

... that said, I've had more practical scenarios where there have been terrain models, and other entourage that needed cleaning or deletion.

I think it should be safe to use this string replacement method with remove_deep2 since the only things that are to be removed either have no inverses, or have an inverse that will also be removed.

If @aothms has a solution where removing 1k elements on a 10MB is no different to removing the same 1k elements on a 200MB file, that would be preferred :)
